service/debugger/debugger.go

```diff
@@ -1467,6 +1467,65 @@ func (d *Debugger) Functions(filter string) ([]string, error) {
 	return funcs, nil
 }
 
+func traverse(t proc.ValidTargets, f *proc.Function, depth int, FollowCalls int) ([]string, error) {
+	if depth > FollowCalls {
+		return nil, nil
+	}
+	funcs := []string{}
+	funcs = append(funcs, f.Name)
+	text, err := proc.Disassemble(t.Memory(), nil, t.Breakpoints(), t.BinInfo(), f.Entry, f.End)
+	if err != nil {
+		return nil, fmt.Errorf("disassemble failed with error %s", err.Error())
+	}
+	depth = depth + 1
+	for _, instr := range text {
```

Comment:
if depth > FollowCalls this loop doesn't do anything (in fact the prior call to Disassemble doesn't do anything either).

Reply:
FollowCalls is the depth till which the user wants the trace to be printed out 
if we are reaching a depth greater than that we simply return hence the code has been written in this way

Reply:
The point I was trying to make is that depth > FollowCalls you can return early, before this loop and the previous Disassemble call.